src/openseadragon.js

```diff
@@ -816,6 +816,32 @@ window.OpenSeadragon = window.OpenSeadragon || function( options ){
                     canvasElement.getContext( '2d' ) );
     }());
 
+    /**
+     * @private
+     * @inner
+     * @function
+     * @returns {Number} The device's pixel density ratio, or null if canvas isn't supported by the browser.
+     */
+    $.pixelDensityRatio = (function () {
+        if ( $.supportsCanvas ) {
```

Comment:
>Is it not already doing this?
Indeed it is! That IIEF pattern is used in lots of places...confused me at times early on :)